extensions/arc/runtime/pom.xml

```diff
@@ -22,6 +22,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-core</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-arc-kotlin</artifactId>
+        </dependency>
```

Comment:
I think this should be added to the `runtime` module of the `kotlin` extension, not the `arc` extension. All Kotlin apps are supposed to use the `kotlin` extension, and it's hard for Quarkus apps to not depend on ArC, so a simple direct dependency should be enough.

Reply:
I was originally contemplating adding it to the context propagation module since there the API is explicit, thus addressing your comment here
https://github.com/quarkusio/quarkus/pull/49057#issuecomment-3106204210

But this sounds better.